vim-patch:8.1.{770,857,914,977,1004,1526,1551,1565,1584,1629,1641,1703,1728,1730,1736,1802,1853,1891,2127,2200},8.2.3922,9.0.{546,928},9.1.1382

janlazo opened the pull request:
vim-patch:8.1.0857: indent functionality is not separated
    
    Problem:    Ignore functionality is not separated.
    Solution:   Move indent functionality into a new file. (Yegappan Lakshmanan,
                closes vim/vim#3886)
    
    https://github.com/vim/vim/commit/4b47162ccede0b6d9cbb9473ad870220a24fbf54
    
    ----
    
    Partial port of v8.1.2127 by porting directly to indent_c.c,
    not indent.c.
 
    ----

    Co-authored-by: Bram Moolenaar <Bram@vim.org>

vim-patch:8.1.0977: blob not tested with Ruby
    
    Problem:    Blob not tested with Ruby.
    Solution:   Add more test coverage.  fixes a crash. (Dominique Pelle,
                closes vim/vim#4036)
    
    https://github.com/vim/vim/commit/0d13cce3453b2274c93c5015faa1993baaebace9
    
    -----
    
    https://github.com/neovim/neovim/pull/15211 ports Blob type.
    "Test_ruby_Vim_blob()" was already committed.
    Skipped because "ruby provider" did not implement Blob API.

    Co-authored-by: Bram Moolenaar <Bram@vim.org>

vim-patch:8.1.1526: no numerical value for the patchlevel
    
    Problem:    No numerical value for the patchlevel.
    Solution:   Add v:versionlong.
    
    https://github.com/vim/vim/commit/37df9a4401f6737d6216306ea77a7e080c942aea
    
    ---
    
    Restore "highest_patch()" solely for "v:versionlong".
    Copy/paste Test_vvar_scriptversion2() from patch 9.1.1540.
    It works without ":scriptversion 2".
    In general, if Vim's test works with ":scriptversion 1", just port it
    for additional coverage.
    
    Co-authored-by: Bram Moolenaar <Bram@vim.org>

vim-patch:8.1.1565: MS-Windows: no sound support
    
    Problem:    MS-Windows: no sound support.
    Solution:   Add sound support for MS-Windows. (Yasuhiro Matsumoto, Ken Takata,
                closes vim/vim#4522)
    
    https://github.com/vim/vim/commit/9b283523f2f75b45feef902b8713808e883d9c19
    
    ----
    
    "sound" feature is N/A now but this updates "v:versionlong" docs.
    
    Co-authored-by: Bram Moolenaar <Bram@vim.org>

vim-patch:8.1.1584: the evalfunc.c file is getting too big
    
    Problem:    The evalfunc.c file is getting too big.
    Solution:   Move channel and job related functions to channel.c.
    
    https://github.com/vim/vim/commit/0a1f56fcfe31be929e9cd8c3d81a984c960e4180
    
    Co-authored-by: Bram Moolenaar <Bram@vim.org>

vim-patch:8.1.1736: viminfo support is spread out
    
    Problem:    Viminfo support is spread out.
    Solution:   Move more viminfo code to viminfo.c. (Yegappan Lakshmanan,
                closes vim/vim#4717)  Reorder code to make most functions static.
    
    https://github.com/vim/vim/commit/c3328169d5566b97a6a6921067017e4369dd7cd6
    
    735aa4c4c89943b26f1d6ba0d3e076002490c09d was the partial port for
    the typedefs.
    This patch completes the viminfo->shada port.
    
    - get_shada_parameter()
    - find_shada_parameter()
    
    Co-authored-by: Bram Moolenaar <Bram@vim.org>

 vim-patch:8.1.2127: the indent.c file is a bit big
    
    Problem:    The indent.c file is a bit big.
    Solution:   Move C-indent code a a new cindent.c file.  Move other
                indent-related code to indent.c. (Yegappan Lakshmanan,
                closes vim/vim#5031)
    
    https://github.com/vim/vim/commit/14c01f83487d5c53192297a710eda2b8a4ab17c9
    
    Co-authored-by: Bram Moolenaar <Bram@vim.org>

------------------

NA patches:

vim-patch:8.1.0770: inconsistent use of ELAPSED_FUNC
    
    Problem:    Inconsistent use of ELAPSED_FUNC.
    Solution:   Consistently use ELAPSED_FUNC.  Also turn ELAPSED_TYPE into a
                typedef. (Ozaki Kiichi, closes vim/vim#3815)
    
    https://github.com/vim/vim/commit/1ac56c2d11da5ffa44db23e1fd0c533d02ab2f66
    
    Co-authored-by: Bram Moolenaar <Bram@vim.org>

vim-patch:8.1.0914: code related to findfile() is spread out
    
    Problem:    Code related to findfile() is spread out.
    Solution:   Put findfile() related code into a new source file. (Yegappan
                Lakshmanan, closes vim/vim#3934)
    
    https://github.com/vim/vim/commit/5fd0f5052f9a312bb4cfe7b4176b1211d45127ee
    
    Co-authored-by: Bram Moolenaar <Bram@vim.org>

vim-patch:8.1.1004: function "luaV_setref()" not covered with tests
    
    Problem:    Function "luaV_setref()" not covered with tests.
    Solution:   Add a test. (Dominique Pelle, closes vim/vim#4089)
    
    https://github.com/vim/vim/commit/e165f63598b58bfdac0168583aa1ef75fbf7be6d
    
    Co-authored-by: Bram Moolenaar <Bram@vim.org>

vim-patch:8.1.1551: warning for shadowing popup_dragwin
    
    Problem:    Warning for shadowing popup_dragwin. (Dominique Pelle)
    Solution:   Add missing change.
    
    https://github.com/vim/vim/commit/6c17543b5634ccccedfbffa4993e9fbd4a990067
    
    Co-authored-by: Bram Moolenaar <Bram@vim.org>

vim-patch:8.1.1629: terminal function help is in the wrong file
    
    Problem:    Terminal function help is in the wrong file.
    Solution:   Move the function details to terminal.txt.
    
    https://github.com/vim/vim/commit/6bf2c6264b5ebbe4981751840c5a8b69da08e744
    
    Co-authored-by: Bram Moolenaar <Bram@vim.org>

vim-patch:8.1.1641: garbage collection may run at a wrong moment
    
    Problem:    Garbage collection may run at a wrong moment. (Trygve Aaberge)
    Solution:   Postpone garbage collection while parsing messages. (closes vim/vim
#4620)
    
    https://github.com/vim/vim/commit/6cc7e21412487ff8bd531c73ac9325f5ba2409a9
    
    Co-authored-by: Bram Moolenaar <Bram@vim.org>
    
vim-patch:8.1.1703: breaking out of loop by checking window pointer insufficient
    
    Problem:    Breaking out of loop by checking window pointer is insufficient.
    Solution:   Check the window ID and the buffer number. (closes vim/vim#4683)
    
    https://github.com/vim/vim/commit/61386408063a2300d7d7f42c5156f66f6252fb54
    
    Co-authored-by: Bram Moolenaar <Bram@vim.org>
    
vim-patch:8.1.1728: wrong place for command line history viminfo support
    
    Problem:    Wrong place for command line history viminfo support.
    Solution:   Move it to viminfo.c.
    
    https://github.com/vim/vim/commit/5f32ece459d1f310b1b48b72e07dcd77d3261a76
    
    Co-authored-by: Bram Moolenaar <Bram@vim.org>
    
vim-patch:8.1.1730: wrong place for mark viminfo support
    
    Problem:    Wrong place for mark viminfo support.
    Solution:   Move it to viminfo.c. (Yegappan Lakshmanan, closes vim/vim#4716)
    
    https://github.com/vim/vim/commit/1e78e69680a5f52970d9b1ef60710e556b09b8c2
    
    Co-authored-by: Bram Moolenaar <Bram@vim.org>

vim-patch:8.1.2200: crash when memory allocation fails
    
    Problem:    Crash when memory allocation fails.
    Solution:   Check for NULL curwin and curbuf. (Christian Brabandt,
                closes vim/vim#4839)
    
    https://github.com/vim/vim/commit/1cac70953d3c012453ea85b1308a8b1f94359c26
    


vim-patch:8.2.3922: cannot build with dynamic Ruby 3.1
    
    Problem:    Cannot build with dynamic Ruby 3.1.
    Solution:   Add "_EXTRA" variables for CI.  Add missing functions. (Ozaki
                Kiichi, closes vim/vim#9420)
    
    https://github.com/vim/vim/commit/8bb3fe4d4dcd27c02e903f6772fdc8fe2e9eba70
    
    Co-authored-by: ichizok <gclient.gaap@gmail.com>
    
vim-patch:9.0.0546: supporting Ruby 1.8 makes code complicated
    
    Problem:    Supporting Ruby 1.8 makes code complicated.
    Solution:   Drop Ruby 1.8 support, it is ancient. (Ken Takata, closes vim/vim#11195)
    
    https://github.com/vim/vim/commit/236ccbf6f8a671a45b773de4b34558ecfdf9ac02
    
    Co-authored-by: K.Takata <kentkt@csc.jp>
    
vim-patch:9.0.0928: using Ruby LDFLAGS may cause build problems
    
    Problem:    Using Ruby LDFLAGS may cause build problems.
    Solution:   Do not add Ruby LDFLAGS to Vim's LDFLAGS. (Zdenek Dohnal,
                closes vim/vim#11592)
    
    Co-authored-by: K.Takata <kentkt@csc.jp>
    
vim-patch:9.0.0928: using Ruby LDFLAGS may cause build problems
    
    Problem:    Using Ruby LDFLAGS may cause build problems.
    Solution:   Do not add Ruby LDFLAGS to Vim's LDFLAGS. (Zdenek Dohnal,
                closes vim/vim#11592)
    
    https://github.com/vim/vim/commit/1d822afaf66a84fab71c84521fd92db11d2c8925
    
    Co-authored-by: Zdenek Dohnal <zdohnal@redhat.com>
    
vim-patch:9.1.1382: if_ruby: unused compiler warnings from ruby internals
    
    Problem:  if_ruby: unused compiler warnings from ruby internals
    Solution: disable -Wunused-parameter for if_ruby internal code
              (Philip H.)
    
    closes: vim/vim#17297
    
    https://github.com/vim/vim/commit/411730e277c408f89c82ae7dbb1ebfa71ec0f4e5
    
    Co-authored-by: Philip H. <47042125+pheiduck@users.noreply.github.com>



------------------

~~lint and clang-analzyer/tidy are in conflict. Or does moving code unsuppress the existing lint warnings?~~ CI passed so it's a matter of adding specific headers in both `.h` and `.c` file. This is going to take awhile with NA patches too since version.c is missing. Since so much time has passed, I'll straight up assume that some Neovim/Vim incompatibilities in async / job / timer/ `:terminal`  in terms of code, docs, runtime cannot be reconciled without significant effort. There's just some patches across Vim `v:version` that should be grouped in a dedicated ticket such that a volunteer will go through all in 1 PR. 

v8.1.770 is for headers that affect `elapsed()` but `elapsed()` is N/A in v8.x patches and Neovim has `uv_now()`.

v8.1.914 is N/A.
- findfile.c -> file_search.c
- misc1.c -> path.c

v8.1.1004 is for testing VIm's if_lua via `test_garbagecollect_now()`. Former is N/A. Latter is debatable since it's part of the public API but the multitude of lua tests running `garbagecollect` should suffice for coverage. 
If python3/ruby/nodejs/perl had to unit-test `test_garbagecollect_now()` and similar debug-testing API, I'll revert the N/A  inclusion.

v8.1.1551 is a hot-fix for v8.1.1525 which could be ported.

v8.1.1629 remaining value is the table of contents for `runtime/doc/terminal.txt` since the only terminal-function is  `nvim_open_term()`, documented in `api.txt`.
These similar "incompatible" runtime patches I'll mark as N/A.
Not explicitly mentioned in https://neovim.io/doc/user/dev_vimpatch.html#_na-(not-applicable)-patches . 

v8.1.{1641, 1703, 2200} are for N/A because of parse_queued_messages().

> if_*.c changes: if_python.c et. al. were removed.

for 8.2.x,9.x compiler patches for `if_ruby`.

-----

Hesitant to mark some features like `sound` as N/A even if 3a1bf826ff62fdfca2444c51f9eca85fade3d020 was merged.
v8.1.1565 was the exception for `v:versionlong`.

janlazo committed:
vim-patch:8.1.0857: indent functionality is not separated

Problem:    Ignore functionality is not separated.
Solution:   Move indent functionality into a new file. (Yegappan Lakshmanan,
            closes vim/vim#3886)

https://github.com/vim/vim/commit/4b47162ccede0b6d9cbb9473ad870220a24fbf54

----

Partial port of v8.1.2127 by porting directly to indent_c.c,
not indent.c.

----

Co-authored-by: Bram Moolenaar <Bram@vim.org>

janlazo committed:
vim-patch:8.1.2127: the indent.c file is a bit big

Problem:    The indent.c file is a bit big.
Solution:   Move C-indent code a a new cindent.c file.  Move other
            indent-related code to indent.c. (Yegappan Lakshmanan,
            closes vim/vim#5031)

https://github.com/vim/vim/commit/14c01f83487d5c53192297a710eda2b8a4ab17c9

Co-authored-by: Bram Moolenaar <Bram@vim.org>

janlazo committed:
vim-patch:8.1.1584: the evalfunc.c file is getting too big

Problem:    The evalfunc.c file is getting too big.
Solution:   Move channel and job related functions to channel.c.

https://github.com/vim/vim/commit/0a1f56fcfe31be929e9cd8c3d81a984c960e4180

---

N/A patches below:

vim-patch:8.1.0770: inconsistent use of ELAPSED_FUNC

Problem:    Inconsistent use of ELAPSED_FUNC.
Solution:   Consistently use ELAPSED_FUNC.  Also turn ELAPSED_TYPE into a
            typedef. (Ozaki Kiichi, closes vim/vim#3815)

https://github.com/vim/vim/commit/1ac56c2d11da5ffa44db23e1fd0c533d02ab2f66

vim-patch:8.1.0914: code related to findfile() is spread out

Problem:    Code related to findfile() is spread out.
Solution:   Put findfile() related code into a new source file. (Yegappan
            Lakshmanan, closes vim/vim#3934)

https://github.com/vim/vim/commit/5fd0f5052f9a312bb4cfe7b4176b1211d45127ee

vim-patch:8.1.1004: function "luaV_setref()" not covered with tests

Problem:    Function "luaV_setref()" not covered with tests.
Solution:   Add a test. (Dominique Pelle, closes vim/vim#4089)

https://github.com/vim/vim/commit/e165f63598b58bfdac0168583aa1ef75fbf7be6d

vim-patch:8.1.1551: warning for shadowing popup_dragwin

Problem:    Warning for shadowing popup_dragwin. (Dominique Pelle)
Solution:   Add missing change.

https://github.com/vim/vim/commit/6c17543b5634ccccedfbffa4993e9fbd4a990067

vim-patch:8.1.1629: terminal function help is in the wrong file

Problem:    Terminal function help is in the wrong file.
Solution:   Move the function details to terminal.txt.

https://github.com/vim/vim/commit/6bf2c6264b5ebbe4981751840c5a8b69da08e744

vim-patch:8.1.1641: garbage collection may run at a wrong moment

Problem:    Garbage collection may run at a wrong moment. (Trygve Aaberge)
Solution:   Postpone garbage collection while parsing messages. (closes vim/vim#4620)

https://github.com/vim/vim/commit/6cc7e21412487ff8bd531c73ac9325f5ba2409a9

vim-patch:8.1.1703: breaking out of loop by checking window pointer insufficient

Problem:    Breaking out of loop by checking window pointer is insufficient.
Solution:   Check the window ID and the buffer number. (closes vim/vim#4683)

https://github.com/vim/vim/commit/61386408063a2300d7d7f42c5156f66f6252fb54

vim-patch:8.1.1802: missing change to call_callback()

Problem:    Missing change to call_callback().
Solution:   Add missing change.

https://github.com/vim/vim/commit/b2129068a55261bea1e293a6a1ce8491a03e7de6

vim-patch:8.1.1853: timers test is still flaky

Problem:    Timers test is still flaky.
Solution:   Compute the time to sleep more accurately.

https://github.com/vim/vim/commit/52953194afccbcb6c2fd013b7a9e2cfbf202b9d0

---
Seems N/A now because of commit 09370eae7756b07bb8f1f116b690b22aec7ec563
---

vim-patch:8.1.2200: crash when memory allocation fails

Problem:    Crash when memory allocation fails.
Solution:   Check for NULL curwin and curbuf. (Christian Brabandt,
            closes vim/vim#4839)

https://github.com/vim/vim/commit/1cac70953d3c012453ea85b1308a8b1f94359c26

vim-patch:8.2.3922: cannot build with dynamic Ruby 3.1

Problem:    Cannot build with dynamic Ruby 3.1.
Solution:   Add "_EXTRA" variables for CI.  Add missing functions. (Ozaki
            Kiichi, closes vim/vim#9420)

https://github.com/vim/vim/commit/8bb3fe4d4dcd27c02e903f6772fdc8fe2e9eba70

vim-patch:9.0.0546: supporting Ruby 1.8 makes code complicated

Problem:    Supporting Ruby 1.8 makes code complicated.
Solution:   Drop Ruby 1.8 support, it is ancient. (Ken Takata, closes vim/vim#11195)

https://github.com/vim/vim/commit/236ccbf6f8a671a45b773de4b34558ecfdf9ac02

vim-patch:9.0.0928: using Ruby LDFLAGS may cause build problems

Problem:    Using Ruby LDFLAGS may cause build problems.
Solution:   Do not add Ruby LDFLAGS to Vim's LDFLAGS. (Zdenek Dohnal,
            closes vim/vim#11592)

https://github.com/vim/vim/commit/1d822afaf66a84fab71c84521fd92db11d2c8925

vim-patch:9.1.1382: if_ruby: unused compiler warnings from ruby internals

Problem:  if_ruby: unused compiler warnings from ruby internals
Solution: disable -Wunused-parameter for if_ruby internal code
          (Philip H.)

closes: vim/vim#17297

https://github.com/vim/vim/commit/411730e277c408f89c82ae7dbb1ebfa71ec0f4e5

Co-authored-by: Bram Moolenaar <Bram@vim.org>

justinmk commented on src/nvim/change.c:
```diff
@@ -957,118 +957,6 @@ int del_bytes(colnr_T count, bool fixpos_arg, bool use_delcombine)
-/// Leaves the cursor on the first non-blank in the line.
-///
-/// @return true if the line was changed.
-bool copy_indent(int size, char *src)
```
this could be `void` if it stayed in this file.

no strong opinion...

janlazo committed:
vim-patch:8.1.0977: blob not tested with Ruby

Problem:    Blob not tested with Ruby.
Solution:   Add more test coverage.  fixes a crash. (Dominique Pelle,
            closes vim/vim#4036)

https://github.com/vim/vim/commit/0d13cce3453b2274c93c5015faa1993baaebace9

-----

https://github.com/neovim/neovim/pull/15211 ports Blob type.
"Test_ruby_Vim_blob()" was already committed.
Skipped because "ruby provider" did not implement Blob API.

Co-authored-by: Bram Moolenaar <Bram@vim.org>

janlazo committed:
vim-patch:8.1.1526: no numerical value for the patchlevel

Problem:    No numerical value for the patchlevel.
Solution:   Add v:versionlong.

https://github.com/vim/vim/commit/37df9a4401f6737d6216306ea77a7e080c942aea

Restore "highest_patch()" solely for "v:versionlong".
Copy/paste Test_vvar_scriptversion2() from patch 9.1.1540.
It works without ":scriptversion 2".
In general, if Vim's test works with ":scriptversion 1", just port it
for additional coverage.

---

vim-patch:8.1.1565: MS-Windows: no sound support

Problem:    MS-Windows: no sound support.
Solution:   Add sound support for MS-Windows. (Yasuhiro Matsumoto, Ken Takata,
            closes vim/vim#4522)

https://github.com/vim/vim/commit/9b283523f2f75b45feef902b8713808e883d9c19

----

"sound" feature is N/A now but this updates "v:versionlong" docs.

Co-authored-by: Bram Moolenaar <Bram@vim.org>

janlazo committed:
vim-patch:8.1.1736: viminfo support is spread out

Problem:    Viminfo support is spread out.
Solution:   Move more viminfo code to viminfo.c. (Yegappan Lakshmanan,
            closes vim/vim#4717)  Reorder code to make most functions static.

https://github.com/vim/vim/commit/c3328169d5566b97a6a6921067017e4369dd7cd6

735aa4c4c89943b26f1d6ba0d3e076002490c09d was the partial port for
the typedefs.
This patch completes the viminfo->shada port.

- get_shada_parameter()
- find_shada_parameter()

Other patches below are N/A.

vim-patch:8.1.1728: wrong place for command line history viminfo support

Problem:    Wrong place for command line history viminfo support.
Solution:   Move it to viminfo.c.

https://github.com/vim/vim/commit/5f32ece459d1f310b1b48b72e07dcd77d3261a76

vim-patch:8.1.1730: wrong place for mark viminfo support

Problem:    Wrong place for mark viminfo support.
Solution:   Move it to viminfo.c. (Yegappan Lakshmanan, closes vim/vim#4716)

https://github.com/vim/vim/commit/1e78e69680a5f52970d9b1ef60710e556b09b8c2

Co-authored-by: Bram Moolenaar <Bram@vim.org>

janlazo committed:
vim-patch:8.1.1891: functions used in one file are global

Problem:    Functions used in one file are global.
Solution:   Add "static". (Yegappan Lakshmanan, closes vim/vim#4840)

https://github.com/vim/vim/commit/5843f5f37b0632e2d706abc9014bfd7d98f7b02e

Co-authored-by: Bram Moolenaar <Bram@vim.org>

janlazo committed:
refactor: fixup! vim-patch:8.1.0877: new buffer

https://github.com/neovim/neovim/pull/9674 removed switch_to_win_for_buf().
This vim-patch removed the last relevant usage of find_win_for_buf()
Vim uses switch_to_win_for_buf() only for if_py_both.

zeertzjq commented on src/nvim/vvars.lua:
```diff
@@ -878,6 +878,21 @@ M.vars = {
+      Like v:version, but also including the patchlevel in the last
+      four digits.  Version 8.1 with patch 123 has value 8010123.
+      This can be used like this: >
+        if v:versionlong >= 8010123
+      <
+      However, if there are gaps in the list of patches included
```
It seems that to match upstream docs, this needs to be changed to this (there is a Tab before the `if`, not two spaces, and there are two Tabs before the "However"):
```suggestion
      	if v:versionlong >= 8010123
      <		However, if there are gaps in the list of patches included
```
as `render_vvar_doc()` uses a line literally if it starts with `>`:
```lua
      if l == '<' or l:match('^<%s') then
        write(l)
      else
        write('\t\t' .. l:gsub('\\<', '<'))
      end
```

janlazo replied:
I grouped the output of `vim-patch.sh -p` of (NA) patches in 1 commit and inserted a note between `Co-author` and patch description.

> The Co-authored-by: line needs to be at the bottom of the commit message, otherwise GitHub won't recognize it.

If you mean to also group these all at the bottom, then sure I can rebase again to update them all.

zeertzjq replied:
Actually, I think the `Co-authored-by:` lines of N/A patches should be removed, and the `Co-authored-by:` line of the non-N/A patch should be moved to the bottom.

zeertzjq replied:
This is not a big deal. Let's just leave as-is.

zeertzjq commented on src/nvim/shada.c:
Should `check_marks_read()` also be moved to shada.c in this commit?

janlazo replied:
Yes. Missed it in the `src/proto/` for viminfo/shada.

zeertzjq commented:
`list_fix_watch()` in patch 8.1.1891 corresponds to `tv_list_watch_fix()` in Nvim. That function can be made static.

zeertzjq commented on test/unit/profile_spec.lua:
```diff
@@ -110,20 +107,6 @@ describe('profiling related functions', function()
     return tonumber(s) + tonumber(us) / 1000000
   end
 
-  describe('profile_equal', function()
-    itp('times are equal to themselves', function()
-      local start = profile_start()
-      assert.is_true(profile_equal(start, start))
-
-      local e = profile_end(start)
-      assert.is_true(profile_equal(e, e))
-    end)
-
-    itp('times are unequal to others', function()
-      assert.is_false(profile_equal(profile_start(), profile_start()))
-    end)
-  end)
-
```
These tests seem still useful. Perhaps it's better to keep `profile_equal()` as non-static instead?

janlazo replied:
No strong opinion. 

```
/// Checks if time `tm1` is equal to `tm2`.
///
/// @return true if `tm1` == `tm2`
static bool profile_equal(proftime_T tm1, proftime_T tm2) FUNC_ATTR_CONST
{
  return tm1 == tm2;
}
```

if that function was more complicated, i may reconsider.
Basic equality check is enough to shim `profile_equal()` outside C codebase.

janlazo commented on src/nvim/eval/buffer.c:
```diff
@@ -706,27 +706,3 @@ void restore_buffer(bufref_T *save_curbuf)
     curbuf->b_nwindows++;
   }
 }
-
-/// Find a window for buffer "buf".
-/// If found true is returned and "wp" and "tp" are set to
-/// the window and tabpage.
-/// If not found, false is returned.
-///
-/// @param       buf  buffer to find a window for
-/// @param[out]  wp   stores the found window
-/// @param[out]  tp   stores the found tabpage
-///
-/// @return  true if a window was found for the buffer.
-bool find_win_for_buf(buf_T *buf, win_T **wp, tabpage_T **tp)
-{
-  *wp = NULL;
-  *tp = NULL;
-  FOR_ALL_TAB_WINDOWS(tp2, wp2) {
-    if (wp2->w_buffer == buf) {
-      *tp = tp2;
-      *wp = wp2;
-      return true;
-    }
-  }
-  return false;
-}
```
It's cleanup for previous patch. Should be considered N/A in patch 8.1.1891

janlazo replied:
commit message will be trimmed more to not fail `lintcommit`. 50 char limit.

zeertzjq replied:
Ah OK

janlazo replied:
I applied your suggestion anyway. Still within limit. Thanks.

janlazo commented:
> And, should vim-patch.sh also rename cindent.c to indent_c.c?

Sure.

janlazo committed:
refactor: fixup! vim-patch:8.2.4881: "P" in Visual

7978660efb087583a1ac49c70588d72af80de6d7 removed all uses of ff:

```diff
-yankreg_T *get_y_previous(void)
-{
-  return y_previous;
-}
-
-void set_y_previous(yankreg_T *yreg)
-{
-  y_previous = yreg;
-}
```

zeertzjq commented:
> Neovim requires VV_KEY to be string as documented, unlike Vim.

This doesn't look correct. In `filter_map_string()`, there is a `vimvars[VV_KEY].vv_type = VAR_NUMBER` assignment, so `VV_KEY` is a number after this line.

Some changes in patch 8.1.1964 depend on patches 8.1.1939 and 8.1.1940 (the changes to how `filter_map()` assigns `vimvars[VV_KEY].vv_type`), so I think it should be ported after these patches.

Also, I noticed that `f_prompt_getprompt()` was added directly to channel.c in [patch 8.2.1588](https://github.com/vim/vim/commit/077cc7aa0e0c431e97795612374fe17fe7c88803), while in Nvim it was added to eval/funcs.c in #14411

janlazo commented:
--- PENDING REVIEW

  * v8.1.1939: code for handling v: variables in generic eval file
  * v8.1.1940: script tests fail
  * v8.1.1943: more code can be moved to evalvars.c
  * v8.1.1954: more functions can be used as a method
  * v8.1.1957: more code can be moved to evalvars.c

  * v8.1.1978: the eval.c file is too big
  * v8.1.1979: code for handling file names is spread out
  * v8.1.1981: the evalfunc.c file is too big
  * v8.1.1982: more functions can be used as methods
  * v8.1.1985: code for dealing with paths is spread out
  * v8.1.1986: more functions can be used as methods
  * v8.1.1987: more functions can be used as methods
  * v8.1.1989: the evalfunc.c file is still too big
  * v8.1.1993: more functions can be used as methods
  * v8.1.1996: more functions can be used as methods

  * v8.1.2021: some global functions can be local to the file
  * v8.1.2022: the option.c file is too big

  * v8.1.2077: the ops.c file is too big

  * v8.1.2092: MS-Windows: redirect in system() does not work
  * v8.1.2093: MS-Windows: system() test fails

  * v8.1.2121: mode is not updated when switching to terminal

  * v8.1.2230: MS-Windows: testing external commands can be improved

  * v8.1.2381: not all register related code is covered by tests

------

> > Neovim requires VV_KEY to be string as documented, unlike Vim.
> 
> This doesn't look correct. In `filter_map_string()`, there is a `vimvars[VV_KEY].vv_type = VAR_NUMBER` assignment, so `VV_KEY` is a number after this line.
> 
> Some changes in patch 8.1.1964 depend on patches 8.1.1939 and 8.1.1940 (which change how `filter_map()` assigns `vimvars[VV_KEY].vv_type`), so I think it should be ported after these patches (in another PR).

You have a point since v8.1.1964 is the final patch among

  * v8.1.1939: code for handling v: variables in generic eval file
  * v8.1.1940: script tests fail
  * v8.1.1943: more code can be moved to evalvars.c
  * v8.1.1954: more functions can be used as a method
  * v8.1.1957: more code can be moved to evalvars.c
  * vim-patch:8.1.1964: crash when using nested map() and filter()

Fine, I'll do it here unless someone volunteers to work on it separately.

> Also, I noticed that `f_prompt_getprompt()` was added directly to channel.c in [patch 8.2.1588](https://github.com/vim/vim/commit/077cc7aa0e0c431e97795612374fe17fe7c88803), while in Nvim it was added to eval/funcs.c in #14411.

Up to you to fixup that patch before or during the port of v8.2.1597. Trying to avoid non v8.1.x patches in this MR (besides N/A).

justinmk commented on runtime/doc/vvars.txt:
```diff
@@ -727,6 +727,19 @@ v:version
 <
 
+				*v:versionlong* *versionlong-variable*
+v:versionlong
```
Do we actually need this? Not a blocker but seems kind of useless.

janlazo replied:
I personally still use `v:version`  so I _may_ (?) migrate to it in the future.

It can be misleading compared to `has('patch-a.b.c')` but it's easier for users who just want to know the highest patch number supported for their configs without resorting  to `:version` or `nvim -v`.

justinmk reviewed (approved):
Nice changes, thanks @janlazo

janlazo commented:
Was going to add last N/A patch for 8.1.0969. 
Waiting for version.c MR to dump the patches.
